### PR TITLE
fix:  unwrap() `None` in NULL value exist multi-field table during prometheus `query_range`

### PR DESCRIPTION
**Pull request by haohuaijin**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

When I use  `curl '127.0.0.1:4004/api/v1/query_range?query=up&start=0&end=100&step=10s'` query a multi-field table which have NULL value. I get a error of `unwrap()` a `Option`. the table look like
```
mysql> select * from up;
+------+------+--------+---------------------+
| host | cpu  | memory | ts                  |
+------+------+--------+---------------------+
| h1   | 66.6 |   1024 | 1970-01-01 08:00:00 |
| h1   | 66.6 |   2048 | 1970-01-01 08:00:02 |
| h1   | 66.6 |   4096 | 1970-01-01 08:00:05 |
| h3   | NULL | 8764.8 | 1970-01-01 08:00:15 |
| h3   | NULL | 754.99 | 1970-01-01 08:00:20 |
| h3   | NULL |   64.8 | 1970-01-01 08:00:30 |
+------+------+--------+---------------------+
```

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

**Review by waynexia — COMMENTED** [review]
Great work, thanks!

File: src/servers/src/prom.rs [hunk 1]
```diff
@@ -350,8 +350,10 @@ impl PromJsonResponse {
...
 
                 // retrieve value
-                let value =
-                    Into::<f64>::into(field_column.get_data(row_index).unwrap()).to_string();
+                let value = match field_column.get_data(row_index) {
+                    Some(v) => Into::<f64>::into(v).to_string(),
+                    None => "NULL".to_string(),
+                };
 
                 buffer.entry(tags).or_default().push((timestamp, value));
```
Comment (waynexia) [inline src/servers/src/prom.rs:358]: It might be better to ignore this data point if the data is null

**Comment by haohuaijin** [comment]
I have question is why we only return the first field of the result🤔？
https://github.com/GreptimeTeam/greptimedb/blob/4151d7a8eae4d7f86bba7becdc0c98fbf2b40885/src/servers/src/prom.rs#L278

**Comment by waynexia** [comment]
> I have question is why we only return the first field of the result🤔？
> 
> https://github.com/GreptimeTeam/greptimedb/blob/4151d7a8eae4d7f86bba7becdc0c98fbf2b40885/src/servers/src/prom.rs#L278

This interface (`src/prom.rs`) you are looking is prometheus-compatible. And prometheus always contains only one field. So other fields are simply dropped here. You might look for [this interface](https://docs.greptime.com/user-guide/prometheus#greptimedb-s-http-api), whose behavior is closer to SQL.

**Review by waynexia — APPROVED** [review]
LGTM 👍

**Review by sunng87 — APPROVED** [review]

**Comment by haohuaijin** [comment]
> > I have question is why we only return the first field of the result🤔？
> > https://github.com/GreptimeTeam/greptimedb/blob/4151d7a8eae4d7f86bba7becdc0c98fbf2b40885/src/servers/src/prom.rs#L278
> 
> This interface (`src/prom.rs`) you are looking is prometheus-compatible. And prometheus always contains only one field. So other fields are simply dropped here. You might look for [this interface](https://docs.greptime.com/user-guide/prometheus#greptimedb-s-http-api), whose behavior is closer to SQL.

Thank you for your reply @waynexia ♥

**Pull request merged**